scripts/dockerfiles/Dockerfile.release

```diff
@@ -16,16 +21,51 @@ COPY Makefile ./
 RUN make plugins && make debug-script
 
 COPY . ./
-RUN make build-linux
+# -buildmode=pie invokes linker incorrectly on arm64 with CGO_ENABLED=0 ?
+RUN --mount=type=cache,target=/root/.cache make build-linux BUILD_MODE=
```

Comment:
We want `pie` for x86, wouldn't this disable it always? It seems to work fine when building images on an actual ARM64 host.

Currently it's an override in the [.circleci/config](https://github.com/aws/amazon-vpc-cni-k8s/blob/fd9e632cc61b18145610a8c8fa4a877d231ee76e/.circleci/config.yml#L52) for `BUILD_MODE`.

Reply:
Yep.  I just disabled it globally here to make the PoC work.  Better solutions are definitely possible.

Note one of the weaknesses of Dockerfile buildx is that it's ugly to do different things for different archs - you need conditionals in reusable shell scripts (eg alplatform) or other onerous solutions.  You can't just set a shell variable in a conditional at the top of Dockerfile, like you would with a normal (single) shell script or Makefile.